Enable authentication between go and mongo db

ydrisR opened the pull request:
* Change mgo source package from labix.org/v2/mgo to gopkg.in/mgo.v2.
    Not sure why it is important as this is suposed to be the same code... but this is the only way I found to make the connection to mongo work (using sasl)
    * change the docker file to build the mgo module with SASL support. This requires the SASL library + a bunch of build tools (gcc)

franck-fourel committed:
Enable travis-ci build (#1)

* Change working folder of travis

ydrisR committed:
Change the mongo driver (mgo) source package

    * Change mgo source package from labix.org/v2/mgo to gopkg.in/mgo.v2.
    Not sure why it is important as this is suposed to be the same code... but this is the only way I found to make the connection to mongo work (using sasl)
    * change the docker file to build the mgo module with SASL support. This requires the SASL library + a bunch of build tools (gcc)

franck-fourel commented on Dockerfile:
```diff
@@ -1,18 +1,22 @@
 # Development
-FROM golang:1.9.2-alpine AS development
+FROM golang:1.10.2-alpine AS development
```
is it necessary to upgrade to 1.10.2? 
Travis is still configured with 1.9.2

ydrisR replied:
Not necessary at all: I simply did that to be aligned with what is done on the platform service.
1.9.2 was release more that a year ago, I supose we should try to keep up to date as much as we can.

franck-fourel committed:
Merge branch 'dblp' into feature/enable-mgo-auth

franck-fourel commented:
I do not squash the merge so that we can cherry pick the ca602e2 to create a pull request on tidepool.

ydrisR committed:
update travis build environment to match the dockerfile target GO 10.2
Also convert test.sh file to unix end of line format to avoid execution issues when running docker on windows
(git on windows seems to automatically modify the end of line character)

ydrisR committed:
correct a few tests which are failling on GO 10.2

ydrisR commented:
@franck-fourel Travis is happy now ;)
